internal/controller/operatorconfigmap_controller.go

```diff
@@ -950,3 +987,59 @@ func (c *OperatorConfigMapReconciler) deleteDelegatedCSI() error {
 	}
 	return nil
 }
+
+func (c *OperatorConfigMapReconciler) getCsiCustomTolerations() ([]corev1.Toleration, []corev1.Toleration, error) {
+	nodePluginTolerations, controllerPluginTolerations := []corev1.Toleration{}, []corev1.Toleration{}
+
+	// list the rook ceph cm
+	rookCephOperatorConfig := &corev1.ConfigMap{}
+	err := c.Client.Get(c.ctx, types.NamespacedName{Name: utils.RookCephOperatorConfigName, Namespace: c.OperatorNamespace}, rookCephOperatorConfig)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			return nodePluginTolerations, controllerPluginTolerations, nil
+		}
+		return nodePluginTolerations, controllerPluginTolerations, fmt.Errorf("failed to get Rook Ceph Operator ConfigMap: %v", err)
+	}
+
+	// Get the CSI tolerations from the Rook Ceph Operator ConfigMap
+	csiPluginTolerations, plugin := rookCephOperatorConfig.Data[utils.CsiPluginTolerationKey]
+	csiProvisionerToleration, provisioner := rookCephOperatorConfig.Data[utils.CsiProvisionerTolerationKey]
+	if (!plugin && !provisioner) || (csiPluginTolerations == "" && csiProvisionerToleration == "") {
```

Comment:
You are already checking for empty values, so there is no point in checking for key existence
 ```suggestion
	if (csiPluginTolerations == "" && csiProvisionerToleration == "") {
```